components/cluster-coordinator/org.wso2.carbon.cluster.coordinator/org.wso2.carbon.cluster.coordinator.rdbms/src/main/java/org/wso2/carbon/cluster/coordinator/rdbms/RDBMSCoordinationStrategy.java

```diff
@@ -35,79 +38,134 @@
+                        getOrDefault(RDBMSConstants.HEART_BEAT_INTERVAL, 1000);
+                // Maximum age of a heartbeat. After this much of time, the heartbeat is considered invalid and node is
+                // considered to have left the cluster.
+                this.heartbeatMaxAge = heartBeatInterval *
+                        (int) strategyConfiguration.getOrDefault(RDBMSConstants.HEART_BEAT_MAX_AGE, 2);
+                this.localGroupId = (String) clusterConfiguration.get(CoordinationPropertyNames.GROUP_ID_PROPERTY);
+            } else {
+                throw new ClusterCoordinationException("Strategy Configurations not found in" +
+                        " deployment.yaml, please check " + CoordinationPropertyNames.STRATEGY_CONFIG_NS + " under "
+                        + CoordinationPropertyNames.CLUSTER_CONFIG_NS + " namespace configurations");
+            }
+        } else {
+            throw new ClusterCoordinationException("Cluster Configurations not found in" +
+                    " deployment.yaml, please check " + CoordinationPropertyNames.CLUSTER_CONFIG_NS +
+                    " namespace configurations");
+        }
+        if (this.localGroupId == null) {
+            throw new ClusterCoordinationException(
+                    "Group Id not set in cluster.config in deployment.yaml configuration file");
+        }
         ThreadFactory namedThreadFactory = new ThreadFactoryBuilder()
                 .setNameFormat("RDBMSCoordinationStrategy-%d").build();
-        this.threadExecutor = Executors.newScheduledThreadPool(
-                CoordinationStrategyConfiguration.getInstance().getRdbmsConfigs()
-                        .get("heartbeatInterval"), namedThreadFactory);
-        this.heartBeatInterval = CoordinationStrategyConfiguration.getInstance().getRdbmsConfigs()
-                .get(CoordinationPropertyNames.RDBMS_BASED_COORDINATION_HEARTBEAT_INTERVAL);
-        // Maximum age of a heartbeat. After this much of time, the heartbeat is considered invalid and node is
-        // considered to have left the cluster.
-        this.heartbeatMaxAge = heartBeatInterval * 2;
-        this.localNodeId = generateRandomId();
-        this.rdbmsMemberEventProcessor = new RDBMSMemberEventProcessor(localNodeId, datasource);
-        this.communicationBusContext = new RDBMSCommunicationBusContextImpl(datasource);
+        this.threadExecutor = Executors.newSingleThreadScheduledExecutor(namedThreadFactory);
+        try {
+            ConfigProvider configProvider = RDBMSCoordinationServiceHolder.getConfigProvider();
+            if (configProvider != null) {
+                this.localNodeId = (String) configProvider.getConfigurationMap("wso2.carbon").get("id");
+            } else {
+                logger.warn("The id of the server has not been set in wso2.carbon namespace of deployment.yaml." +
+                        " Some features may not work correctly unless this configuration is specified.");
```

Comment:
What features will not work?

Reply:
For the dashboard it would also take this configuration as the node id

Reply:
But since this is a component u should only speak from the component's perspective.
means -> what effect it has on ur component.
Since it's a warn IMO u should tell something like;
x config is missing in deployment.yaml hence using an auto-generated value of y as the x.

Reply:
Message changed to give more meaning